compiler/rustc_hir_typeck/src/upvar.rs

```diff
@@ -1161,7 +1163,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
 
         let ty = match closure_clause {
             hir::CaptureBy::Value { .. } => ty, // For move closure the capture kind should be by value
-            hir::CaptureBy::Ref => {
+            hir::CaptureBy::Ref | hir::CaptureBy::Use { .. } => {
```

Comment:
Hmm, I'm not sure about this.

Reply:
This is specific to 2229 migrations, which is kind of weird. I wonder if we should limit `use` to Rust 2021 just to avoid having to think about it.

Reply:
I was also wondering about editions here, my code is only using the feature flag but yeah maybe makes sense to use the edition as this is going to land at some point in an edition anyway, right?.